openmc/data/library.py

```diff
@@ -22,6 +22,12 @@ class DataLibrary(EqualityMixin):
     def __init__(self):
         self.libraries = []
 
+    def __getitem__(self, material):
+        for library in self.libraries:
+            if material in library['materials']:
+                return library
+        return None
```

Comment:
I'm guilty of this too, but the risk of having this as a special method is that no users will ever know about it. It might be better to just have this as a normal method with an accompanying docstring.